snooty/postprocess.py

```diff
@@ -615,6 +616,59 @@ def add_titles_to_label_targets(
                     target.children = heading.children
             self.pending_targets = []
 
+    def bound_included_AST(
```

Comment:
I feel like this function is trying to reinvent the wheel: the purpose of the Postprocessor class is to iterate through a page's AST and perform an operation on it. Instead of writing a _new_ function to dive through the children, a new handler class could perform the work of identifying start-after/end before nodes as the event parser moves through them. Happy to chat more to flesh this out!